src/memory.h

```diff
@@ -0,0 +1,26 @@
+/* SPDX-License-Identifier: GPL-2.0 OR BSD-2-Clause */
+/*
+ * Copyright 2022 Amazon.com, Inc. or its affiliates. All rights reserved.
+ */
+
+#ifndef MEMORY_H
+#define MEMORY_H
+
+#include <stddef.h>
+#include <stdint.h>
+#include <stdbool.h>
+
+#define container_of(ptr, type, member) ({                      \
+                const typeof( ((type *)0)->member ) *__mptr = (ptr);    \
+                (type *)( (char *)__mptr - offsetof(type,member) );})
+
+
+/* Base context for memory management to be extended by concrete implementations */
+struct memory_ctx {
```

Comment:
Why not keep the ctx struct separate from the ops with an ops struct that gets initialized for each accelerator type?

Reply:
We can do this but ctx struct is specific for each memory type so in perftest_parameters.c we will need allocate and init both ctx and ops for every memory type, and this I would try to avoid. What advantage do you see in holding them separately?